src/cpp/session/modules/ModuleTools.R

```diff
@@ -358,4 +358,69 @@
    invisible(output)
 })
 
+# create an environment to hold original versions of S3 functions we have overridden
+assign(".rs.S3Originals", new.env(parent = emptyenv()), envir = .rs.toolsEnv())
 
+if (getRversion() < "3.5") {
```

Comment:
Do the objects created by running this code get cached somewhere (and then reloaded for subsequent sessions)? (Just want to make sure that it's not possible for someone to end up with R 3.5 components when running with R 3.4)

Reply:
No, I don't believe so. We source the module .R files, including this one, at the beginning of every session, whether new or resumed.